### DISCUSSION TITLE
Execution based extension recommendations

### DISCUSSION THREAD

**Discussion by radeksimko** [microsoft/vscode-discussions#359] [Extension Development QnA] [answered]
There is a number of extensions which seem to be recommended based on an installed application. This is typically also indicated on the extension page too with the following message:

> This extension is recommended because you have XYZ installed.

The examples I was able to find involve Docker and Kubernetes, but I was unable to find any documentation or details on where do these recommendations come from.

![Screenshot 2023-01-01 at 17 52 59](https://user-images.githubusercontent.com/287584/210180362-e542eaf2-d1e4-41ba-9083-b487ad12b3d4.png)
![Screenshot 2023-01-01 at 17 53 52](https://user-images.githubusercontent.com/287584/210180364-edb20089-7f76-4620-b06f-dad1c4a3489a.png)

TL;DR Who/what controls which exact extensions come up as recommended when e.g. `docker` or `kubectl` executable is found in the appropriate location?

I was expecting to find something in the `package.json` of the above extensions, but nothing obvious came up, which leads me to believe these recommendations are managed somewhere behind the scenes by Microsoft staff?

All I can find is that it was originally introduced as part of https://github.com/microsoft/vscode/issues/101903 / https://github.com/microsoft/vscode/commit/be48f1d5e84ffd44bae71191f0399289bccda902 by @sandy081

Related question - what is the process for adding such a recommendation? Specifically I'm curious about the possibility of recommending https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform (official HashiCorp Terraform extension) based on `terraform` or `terraform.exe` (i.e. Terraform CLI as installed via official installation channels per https://developer.hashicorp.com/terraform/downloads)

**Comment by sandy081** [comment]
CC @isidorn

**Comment by isidorn** [comment, marked as answer]
@radeksimko we control these recommendations as part of the VS Code product. Each recommendation we review carefully before we add it.
Recommending the terraform extension when terraform execution is there sounds reasonable. Can you please send me an email to inikolic@microsoft.com and we can discuss this further. Thanks

**Reply by radeksimko** [reply]
Done, sent you an email. Thanks!

**Reply by sxkm** [reply]
Hi

**Comment by FLUXparticle** [comment]
How would I in general ask for a recommendation of my extension?
